src/services/get-env-api.js

```diff
@@ -1,8 +1,24 @@
 import store from "../store";
 import constants from "../utils/constants";
 
-export default function () {
-  return store.getters.env === "test"
-    ? constants.api.testnet
-    : constants.api.mainnet;
+const CURRENT_API_VERSION = import.meta.env.VITE_CURRENT_API_VERSION;
+
+export default function (version) {
+  let apiEndpoint =
+    store.getters.env === "test"
+      ? constants.api.testnet
+      : constants.api.mainnet;
+
+  if (version === false) {
+    return apiEndpoint;
+  }
+
+  if (typeof version !== "string" || !version.trim()) {
```

Comment:
if **version** is undefined, `.trim()` will throw error.

Reply:
Yeah, so actually if first condition is true then it won't check second condition. So in case version is undefined, first condition becomes true and `version.trim()` will only be checked if version is a string

Reply:
okay

Reply:
The condition in the if statement here is a bit difficult to follow. Trimming a string is an effect and we shouldn't rely on implicitly typecasting its return value.

If I understand the intent correctly, can we rephrase to:

```js
version = typeof version === "string"
  ? version.trim()
  : CURRENT_API_VERSION
```

Reply:
I guess this check is not needed, if the version has default value @ajithranka